src/app/services/pad.service.ts

```diff
@@ -42,25 +42,32 @@ export class PadService {
    * with this service. Hence, changes in session would happen
    * after destroying the view and cleaning the service.
    *
-   * @param objectId
+   * @param name
    */
-  init(objectId) {
-
-    if (this.editor.hasDocument()) {
-      this.editor.clean();
-    }
-
-    if (this.session) {
-      this.loadObject(objectId);
-    } else {
-      // we must wait for the swell session
-      this.swellService.session$.pipe(first( val => val != null))
-      .subscribe(session => {
-          this.session = session;
-          this.loadObject(objectId);
-      });
-
-    }
+  init(name) {
+    // get the id from the name
+    this.swellService.getObjectNames(name)
```

Comment:
Would be good to encapsulate the "get object id from name" logic in a method of swellService?